### DISCUSSION TITLE
Collection seems to be corrupt, how do I fix it?

### DISCUSSION THREAD

**Discussion by ammaraga** [directus/directus#13634] [Archive]
I did everything the same as with other collections (relationships), but one collection is broken. Directus shows "You don't have permission to access this." when I try to show content.  
Data model doesn't show any fields whatsoever, I only have an option, which I can't, because I don't want to have that permission for Directus. Any idea how to fix this collection?
I've also tried deleting the records in dbo.directus_collections and dbo.directus_fields. No effect.

This the T-SQL grant statement:
GRANT select, insert, update, delete, alter ON dbo.table TO svc_cms

**Comment by rijkvanzanten** [comment]
It's impossible to accurately tell you what's wrong or how to fix it without some more information. Can you share a database dump with a reproduction?

**Comment by sxsws** [comment]
I've seen this error a lot and it means that there's a record in one of the directus tables that doesn't exist in the table (or perhaps vice versa). I'm going to guess from your description there's a record in `directus_relations` that points to a non-existant relation, or the record is wrong for the existing relation. If you play with the table manually, change the `directus_{fields,collections,relations}` tables as necessary.

**Reply by ammaraga** [reply]
Nope, I've checked all the dbo.directus_* tables, I made sure there was no reference to the problematic collection/table, I revoked the permissions. But as I've tried to recreate the collection, no fields are shown in the model area, and when trying to show the data, the error       "message": "You don't have permission to access this."," is shown.

**Reply by sxsws** [reply]
check for reverse relations pointing to it

**Reply by ammaraga** [reply]
relations table is completely empty

**Reply by sxsws** [reply]
does your collection have relations to other tables though?

**Reply by ammaraga** [reply]
I´ve deleted the relations few days ago, so now I have very basic setup.
